client/state/data-layer/wpcom/sites/jitm/index.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { isJetpackSite } from 'state/sites/selectors';
+import { JITM_SET, SECTION_SET, SELECTED_SITE_SET } from 'state/action-types';
+import config from 'config';
+import { makeParser } from 'state/data-layer/wpcom-http/utils';
+import schema from './schema.json';
+
+/**
+ * Poor man's process manager
+ * @type {{
+ * hasInitializedSites: boolean,
+ * hasInitializedSection: boolean,
+ * lastSection: null|string,
+ * lastSite: null|string,
+ * }}
+ */
+const process = {
+	hasInitializedSites: false,
+	hasInitializedSection: false,
+	lastSection: null,
+	lastSite: null,
+};
+
+const unescapeDecimalEntities = ( str ) => {
+	return str.replace( /&#(\d+);/g, ( match, entity ) => String.fromCharCode( entity ) );
+};
+
+const transformApiRequest = ( { data: jitms } ) => jitms.map( ( jitm ) =>
+	( {
+		message: unescapeDecimalEntities( jitm.content.message ),
+		description: unescapeDecimalEntities( jitm.content.description ),
+		featureClass: jitm.feature_class,
+		callToAction: unescapeDecimalEntities( jitm.CTA.message ),
+		id: jitm.id,
+	} )
+);
+
+const insertJITM = ( dispatch, siteId, messagePath, jitms ) =>
+	dispatch( {
+		type: JITM_SET,
+		keyedPath: messagePath + siteId,
+		jitms: jitms.map( jitm => (
+			{ ...jitm, lastUpdated: Date.now() }
+		) )
+	} );
```

Comment:
Note that if we re-write the function signature here we can use this function directly in `dispatchRequest()` instead of wrapping with the `receive` and `failed` helpers